pkg/monitortests/metrics/rate_analyzer.go

```diff
@@ -0,0 +1,55 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	prometheustypes "github.com/prometheus/common/model"
+)
+
+// RateSeriesAnalyzer analyzes a rate bases timeseries
+type RateSeriesAnalyzer struct{}
+
+func (RateSeriesAnalyzer) Analyze(ctx context.Context, query QueryRunner, start, end time.Time, callback Callback) error {
+	result, err := query.RunQuery(ctx, start, end)
+	if err != nil {
+		return fmt.Errorf("query returned error, monitor: %s, err: %w", callback.Name(), err)
+	}
+	if result.Type() != prometheustypes.ValMatrix {
+		return fmt.Errorf("expected a prometheus Matrix type, monitor: %s", callback.Name())
+	}
+
+	zero := prometheustypes.SampleValue(0)
+	matrix := result.(prometheustypes.Matrix)
```

Comment:
should we check for `nil`/`empty` responses ?
why not to do a type assertion ? (not sure if line `19` is enough)